event-service/src/main/java/hcmus/alumni/event/controller/EventServiceController.java

```diff
@@ -0,0 +1,254 @@
+package hcmus.alumni.event.controller;
+
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+import hcmus.alumni.event.dto.IEventDto;
+import hcmus.alumni.event.model.EventModel;
+import hcmus.alumni.event.model.StatusPostModel;
+import hcmus.alumni.event.model.FacultyModel;
+import hcmus.alumni.event.model.UserModel;
+import hcmus.alumni.event.repository.EventRepository;
+import hcmus.alumni.event.utils.ImageUtils;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.PersistenceContext;
+import jakarta.persistence.TypedQuery;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Order;
+import jakarta.persistence.criteria.Predicate;
+import jakarta.persistence.criteria.Root;
+import jakarta.persistence.criteria.Selection;
+
+@RestController
+@CrossOrigin(origins = "http://localhost:3000")
+@RequestMapping("/event")
+public class EventServiceController {
+	@PersistenceContext
+	private EntityManager em;
+	@Autowired
+	private EventRepository eventRepository;
+	@Autowired
+    private ImageUtils imageUtils;
+	
+	@GetMapping("/")
```

Comment:
thêm request param search theo khoa đi ô
[tham khảo](https://stackoverflow.com/questions/70165854/spring-data-jpa-how-to-form-a-conditional-query)